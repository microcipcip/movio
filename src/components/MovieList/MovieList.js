import React from 'react'
import PropTypes from 'prop-types'
import Movie from 'components/Movie'

const MovieList = props => [1, 2, 3, 4, 5].map(n => <Movie key={n} n={n} />)

MovieList.propTypes = {}
MovieList.defaultProps = {}

export default MovieList
