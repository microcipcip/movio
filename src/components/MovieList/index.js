import Index from './MovieList'

export default Index
