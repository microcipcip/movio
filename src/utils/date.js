export const addZero = date => ('0' + date).slice(-2)

export const formatMovieDate = movieDate => {
  const d = new Date(movieDate)
  return `${addZero(d.getDate())} ${addZero(
    d.getMonth() + 1
  )} ${d.getFullYear()}`
}
