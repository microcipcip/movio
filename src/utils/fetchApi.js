// wrapper for the fetch API
// because it doesn't throw errors when response status is 400
import { TMDB_API_KEY, TMDB_API_URL, TMDB_API_LANG } from 'config/index'

const parseJSON = response =>
  new Promise(resolve =>
    response.json().then(json =>
      resolve({
        status: response.status,
        ok: response.ok,
        json,
      })
    )
  )

export default url => {
  const queryOrAmpersand = url.includes('?') ? '&' : '?'
  const query = `${TMDB_API_URL}${url}${queryOrAmpersand}api_key=${TMDB_API_KEY}&language=${TMDB_API_LANG}`

  return new Promise((resolve, reject) =>
    fetch(query)
      .then(parseJSON)
      .then(response => {
        if (response.ok) {
          return resolve(response.json)
        }
        // extract the error from the server's json
        return reject(response.json.meta.error)
      })
      .catch(err =>
        reject({
          networkError: err.message,
        })
      )
  )
}
