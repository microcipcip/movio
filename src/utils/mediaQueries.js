import { css } from 'styled-components'

// Breakpoint
export const mediaWidthRule = rule =>
  `${{ up: 'min', down: 'max' }[rule] || 'min'}-width`

export const ruleTemplate = (rule, width) => `(${rule}: ${width}px)`
export const mediaTemplate = rules => `@media only screen and ${rules}`

export const getSmallestMedia = breakpoints => {
  const keys = Object.keys(breakpoints)

  return Number(
    keys.reduce((acc, key) => {
      if (acc) {
        return acc > breakpoints[key] ? breakpoints[key] : acc
      }
      return breakpoints[key]
    }, undefined)
  )
}

export const getNextMedia = (breakpoints, width) => {
  const keys = Object.keys(breakpoints)

  const nextBreakpoint = Number(
    keys.reduce((acc, key) => {
      const current = breakpoints[key]
      if (acc && width - current <= 0) {
        return width - acc > Math.abs(width - current) || width - acc === 0
          ? current
          : acc
      }
      return getSmallestMedia(breakpoints)
    }, undefined)
  )

  if (nextBreakpoint === width) {
    throw new Error(
      `The breakpoint of value ${nextBreakpoint} is the heighst there is, did you mean to use "[media].up"?`
    )
  }
  return nextBreakpoint
}

export const mediaRules = (breakpoints, widthKey, rule, boundKey) => {
  const width = breakpoints[widthKey]
  const bound = breakpoints[boundKey]
  let baseWidthRule = mediaWidthRule(rule)
  let boudWidthRule

  let baseRule = ruleTemplate(baseWidthRule, width)
  let boundRule

  if (bound && width) {
    // Get correct rule based on width relative to bound
    baseWidthRule = mediaWidthRule(bound <= width ? 'down' : 'up')
    boudWidthRule = mediaWidthRule(bound <= width ? 'up' : 'down')

    baseRule = ruleTemplate(
      mediaWidthRule(bound <= width ? 'down' : 'up'),
      width
    )
    boundRule = ruleTemplate(boudWidthRule, bound)
    return [].concat([baseRule], bound ? [boundRule] : []).join(' and ')
  }

  if (!bound && rule === 'only') {
    // Get correct rule based on width relative to bound
    boudWidthRule = mediaWidthRule(bound <= width ? 'up' : 'down')
    boundRule = ruleTemplate(boudWidthRule, getNextMedia(breakpoints, width))
  }

  return [].concat([baseRule], boundRule ? [boundRule] : []).join(' and ')
}

export const getMixin = (breakpoints, key, rule = 'up') => bound => (
  ...args
) => css`
  ${mediaTemplate(mediaRules(breakpoints, key, rule, bound))} {
    ${css(...args)}
  }
`

export const getMediaShorthands = (breakpoints, rule) =>
  Object.keys(breakpoints).reduce(
    (acc, key) => ({
      ...acc,
      [key]: getMixin(breakpoints, key, rule)(),
    }),
    {}
  )

export const getMedia = breakpoints => {
  const mediasUp = getMediaShorthands(breakpoints, 'up')
  const list = Object.keys(breakpoints)

  return {
    ...mediasUp,
    up: widthKey => getMixin(breakpoints, widthKey, 'up')(),
    down: widthKey => getMixin(breakpoints, widthKey, 'down')(),
    only: (widthKey, boundKey) =>
      getMixin(breakpoints, widthKey, 'only')(boundKey),
    list,
  }
}

export default getMedia
