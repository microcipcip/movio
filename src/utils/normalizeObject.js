export default (acc, cur) => {
  acc[cur.id] = cur
  return acc
}
