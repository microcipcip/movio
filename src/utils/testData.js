export const genreList = {
  1: {
    id: 1,
    name: 'Action',
    isChecked: false,
  },
  2: {
    id: 2,
    name: 'Adventure',
    isChecked: false,
  },
  3: {
    id: 3,
    name: 'Animation',
    isChecked: false,
  },
  4: {
    id: 4,
    name: 'Comedy',
    isChecked: false,
  },
  5: {
    id: 5,
    name: 'Crime',
    isChecked: false,
  },
}

export const movieList = [
  {
    id: 1,
    vote_average: 6.5,
    title: 'Venom',
    popularity: 423.154,
    poster_path: '/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg',
    genre_ids: [1, 2],
    release_date: '2018-10-03',
  },
  {
    id: 2,
    vote_average: 7,
    title: 'Fantastic Beasts: The Crimes of Grindelwald',
    popularity: 220.606,
    poster_path: '/uyJgTzAsp3Za2TaPiZt2yaKYRIR.jpg',
    genre_ids: [3, 4],
    release_date: '2018-11-14',
  },
  {
    id: 3,
    vote_average: 8.2,
    title: 'Bohemian Rhapsody',
    popularity: 211.565,
    poster_path: '/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg',
    genre_ids: [5],
    release_date: '2018-10-24',
  },
  {
    id: 4,
    vote_average: 5.9,
    title: 'Mortal Engines',
    popularity: 201.549,
    poster_path: '/uXJVpPXxZO4L8Rz3IG1Y8XvZJcg.jpg',
    genre_ids: [1],
    release_date: '2018-12-06',
  },
  {
    id: 5,
    vote_average: 6,
    title: 'Robin Hood',
    popularity: 171.519,
    poster_path: '/AiRfixFcfTkNbn2A73qVJPlpkUo.jpg',
    genre_ids: [2],
    backdrop_path: '/AuA50D7T7S7OEVcGo0ZKaMhERn0.jpg',
    release_date: '2018-11-20',
  },
  {
    id: 6,
    vote_average: 5.9,
    title: 'The Seven Deadly Sins: Prisoners of the Sky',
    popularity: 150.182,
    poster_path: '/r6pPUVUKU5eIpYj4oEzidk5ZibB.jpg',
    genre_ids: [3],
    release_date: '2018-08-18',
  },
  {
    id: 7,
    vote_average: 7.5,
    title: 'Ralph Breaks the Internet',
    popularity: 141.193,
    poster_path: '/m110vLaDDOCca4hfOcS5mK5cDke.jpg',
    genre_ids: [4, 5],
    release_date: '2018-11-20',
  },
]
