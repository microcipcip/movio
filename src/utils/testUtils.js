import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import 'jest-styled-components'
import 'jest-dom/extend-expect'
import { genreList, movieList } from 'utils/testData'
import * as s from 'styles/vars'

import CinemaContext, { cinemaContextState } from 'components/App/CinemaContext'

const customRender = (node, data = {}) => {
  return render(
    <CinemaContext.Provider value={Object.assign(cinemaContextState, data)}>
      <ThemeProvider theme={s.theme.dark}>{node}</ThemeProvider>
    </CinemaContext.Provider>
  )
}

const domToString = el => {
  const clnEl = el.cloneNode(true)
  const tmp = document.createElement('div')
  tmp.appendChild(clnEl)
  return tmp.innerHTML
}

// re-export everything
export * from 'react-testing-library'
export { customRender, domToString, genreList, movieList }
