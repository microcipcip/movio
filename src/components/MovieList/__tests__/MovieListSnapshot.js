import React from 'react'
import { customRender, genreList, movieList } from 'utils/testUtils'
import MovieList from '../MovieList'

it('Should create snapshot', async () => {
  const { container } = customRender(<MovieList />, { genreList, movieList })
  expect(container).toMatchSnapshot()
})
