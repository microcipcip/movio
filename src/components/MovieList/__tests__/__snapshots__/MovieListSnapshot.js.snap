// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create snapshot 1`] = `
.c2 {
  position: relative;
}

.c2:after {
  content: '';
  display: block;
  padding-top: 150%;
}

.c3 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 2px;
}

.c5 {
  margin: 0;
  font-weight: bold;
  font-size: 1.8rem;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  margin: 0.5em 0 0;
  font-weight: bold;
  font-size: 1.8rem;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin: 0.1em 0.4em 0.1em 0;
  color: #555555;
  font-size: 1.2rem;
}

.c7:after {
  content: ', ';
}

.c7:last-child:after {
  display: none;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 {
  display: grid;
  grid-template-columns: repeat(auto-fill,minmax(140px,2fr));
  grid-gap: 16px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 10px 0 0;
}

@media only screen and (min-width:768px) {
  .c0 {
    grid-template-columns: repeat(auto-fill,minmax(200px,2fr));
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="Venom"
          class="c3"
          src="https://image.tmdb.org/t/p/w342/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg"
        />
      </div>
      <div
        class="typ c4"
      >
        <h2
          class="c5"
        >
          Venom
        </h2>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Action
          </span>
          <span
            class="c7"
          >
            Adventure
          </span>
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="Fantastic Beasts: The Crimes of Grindelwald"
          class="c3"
          src="https://image.tmdb.org/t/p/w342/uyJgTzAsp3Za2TaPiZt2yaKYRIR.jpg"
        />
      </div>
      <div
        class="typ c4"
      >
        <h2
          class="c5"
        >
          Fantastic Beasts: The Crimes of Grindelwald
        </h2>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Animation
          </span>
          <span
            class="c7"
          >
            Comedy
          </span>
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="Bohemian Rhapsody"
          class="c3"
          src="https://image.tmdb.org/t/p/w342/lHu1wtNaczFPGFDTrjCSzeLPTKN.jpg"
        />
      </div>
      <div
        class="typ c4"
      >
        <h2
          class="c5"
        >
          Bohemian Rhapsody
        </h2>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Crime
          </span>
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="Mortal Engines"
          class="c3"
          src="https://image.tmdb.org/t/p/w342/uXJVpPXxZO4L8Rz3IG1Y8XvZJcg.jpg"
        />
      </div>
      <div
        class="typ c4"
      >
        <h2
          class="c5"
        >
          Mortal Engines
        </h2>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Action
          </span>
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="Robin Hood"
          class="c3"
          src="https://image.tmdb.org/t/p/w342/AiRfixFcfTkNbn2A73qVJPlpkUo.jpg"
        />
      </div>
      <div
        class="typ c4"
      >
        <h2
          class="c5"
        >
          Robin Hood
        </h2>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Adventure
          </span>
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="The Seven Deadly Sins: Prisoners of the Sky"
          class="c3"
          src="https://image.tmdb.org/t/p/w342/r6pPUVUKU5eIpYj4oEzidk5ZibB.jpg"
        />
      </div>
      <div
        class="typ c4"
      >
        <h2
          class="c5"
        >
          The Seven Deadly Sins: Prisoners of the Sky
        </h2>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Animation
          </span>
        </div>
      </div>
    </div>
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="Ralph Breaks the Internet"
          class="c3"
          src="https://image.tmdb.org/t/p/w342/m110vLaDDOCca4hfOcS5mK5cDke.jpg"
        />
      </div>
      <div
        class="typ c4"
      >
        <h2
          class="c5"
        >
          Ralph Breaks the Internet
        </h2>
        <div
          class="c6"
        >
          <span
            class="c7"
          >
            Comedy
          </span>
          <span
            class="c7"
          >
            Crime
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
